examples/benchmark/README.md

```diff
@@ -0,0 +1,163 @@
+training pipelines. This way, one can mix and match these configurations and observe the impact of different components
+to
+the end to end training performance. To achieve this, we split configuration to 4 pieces,
+namely; `dataset`, `dataloader`, `checkpoint`, `training`.
+
+The `dataset` configuration keeps information about where data resides. While we support sharded objects, only loading
+from TAR objects supported currently. See [a](#benchmarking-s3-connector-for-pytorch)
+The `dataloader` configuration contains dataloading specific setup. `kind` specify which PyTorch dataset to
+use (`s3iterabledataset`, `s3mapdataset`, `fsspec`) while
+`batch_size`, `num_workers` specify
+relevant [PyTorch DataLoader](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html) parameters. `training`
+configuration specify what model to learn and how many epochs to execute the training for. Currently, we support two
+models `Entitlement` and [ViT](https://huggingface.co/docs/transformers/model_doc/vit).
+To make it easier to add new models, and abstract the learning-sample processing logic from configuration, this module
+defines a Model interface where each model expected to implement
+`load_sample`, `train`, and `save` methods. Lastly, `checkpoint` configuration, defines where and how frequently to save
+the checkpoints.
+
+Once the sub-configurations are defined, one can easily create experimental configuration, that will use Hydra Sweeper
+to launch multiple experiments, sequentially, and monitor performance.
+For example, dataloading experiment configuration, is stored at configuration/dataloading.yaml, has the following
+content:
+
+```
+defaults:
+  - dataloader: ???
+  - dataset: 20k_imagenet
+  - training: entitlement
+  - checkpoint: none
+  - _self_
+
+
+hydra:
+  mode: MULTIRUN
+
+  sweeper:
+    params:
+      dataloader: s3iterabledataset, fsspec
+      dataloader.num_workers: 2,4,8,16
+```
+
+This configuration, puts a pin to dataset and training model while changing dataloader parameters to change `kind`
+and `number_workers`. Once executed with this configuration, benchmark with sequentially execute 8 experiments,
+with the different combinations of swept parameters. As `Entitlement` is not really performing any training, this
+experiments is helpful to see upper-limit of dataloader throughput without having the backpressure from GPU.
+
+## Getting Started
+
+To get started, launch an EC2 instance with a GPU (we used
+a [g5.2xlarge](https://aws.amazon.com/ec2/instance-types/g5/)),
+choosing
+the [AWS Deep Learning AMI GPU PyTorch 2.0.1 (Amazon Linux 2)](https://aws.amazon.com/releasenotes/aws-deep-learning-ami-gpu-pytorch-2-0-amazon-linux-2/)
+as your AMI.
+
+You will need an S3 Bucket with training data and update the `dataset` configuration with correct prefix and region.
+
+Then from this directory, install the dependencies for the example code:
+
+    python -m pip install -r requirements.txt
```

Comment:
nit: Should we add virtual environment creation comments too?

Reply:
Probably best. Will include it.